smarts/core/agent_manager.py

```diff
@@ -162,37 +168,39 @@ def observe_from(
         return observations, rewards, scores, dones
 
     def observe(
-        self, sim
+        self,
     ) -> Tuple[
         Dict[str, Union[Dict[str, Observation], Observation]],
         Dict[str, Union[Dict[str, float], float]],
         Dict[str, Union[Dict[str, float], float]],
         Dict[str, Union[Dict[str, bool], bool]],
     ]:
         """Generate observations from all vehicles associated with an active agent."""
+        sim = self._sim()
+        assert sim
```

Comment:
`self._sim` should always be alive if we're executing code in the agent manger, which is why I just assert that whenever it's used.  

The point of adding `weakrefs` throughout was to avoid the circular reference loop -- like here,  `SMARTS` holding a strong reference to the agent manager object, and the agent manager object holding a strong reference to the SMARTS object.  Python allows these, but it can sometimes delay garbage collection.  In this particular case, it's not really that important since there's only one agent manager object and its life cycle is usually the same as the life cycle of the SMARTS object.  So I would say that if the code above is jarring, we can revert it.

I would actually prefer reverting it to adding a function for the above two lines of code which would then return the `sim` object, because it would basically look identical to the above!  So I'm happy to revert it here if you prefer that -- your call!